test/index.test.ts

```diff
@@ -93,6 +93,36 @@ test("throws if invalid 'type' is provided", async () => {
   );
 });
 
+test("throws if invalid Private Key is provided", async () => {
+  const auth = createAppAuth({
+    appId: APP_ID,
+    privateKey: "INVALID_PRIVATE_KEY",
+  });
+
+  const expectedError = new Error();
+  Object.assign(expectedError, {
+    code: "ERR_OSSL_PEM_NO_START_LINE",
+    function: "get_name",
+    library: "PEM routines",
+    reason: "no start line",
+  });
+
+  await expect(auth({ type: "app" })).rejects.toEqual(expectedError);
+});
+
+test("throws if incomplete Private Key is provided", async () => {
+  const auth = createAppAuth({
+    appId: APP_ID,
+    privateKey: "-----BEGIN RSA PRIVATE KEY-----",
+  });
+
+  await expect(auth({ type: "app" })).rejects.toEqual(
+    new Error(
+      "Private key is incomplete. Make sure it is a single line String and newlines have been replaced by '\n'"
```

Comment:
Looks good! I suggest we slightly change the error message to

```suggestion
      "The 'privateKey` option contains only the first line '-----BEGIN RSA PRIVATE KEY-----'. If you are setting it using a `.env` file, make sure it is set on a single line with newlines replaced by '\n'"
```

what do you think?